test/utils/wasm3_engine.cpp

```diff
@@ -84,14 +84,13 @@ bool Wasm3Engine::instantiate(bytes_view wasm_binary)
     // Transfers ownership to runtime.
     if (m3_LoadModule(m_runtime, module) != m3Err_none)
     {
-        m3_FreeModule(module);
+        // NOTE: apparently m3_FreeModule isn't needed in neither the failure nor the success case
```

Comment:
How this is known?

Reply:
By going through the wasm3 code. Basically it seem once a module is parsed you have two choices: `LoadModule` or `FreeModule`, and if `LoadModule` was called then it is the liability of that to deal with it. Started opening in issue on their repo to clarify this, but then I guess forgot to submit.

Reply:
Asked the question: https://github.com/wasm3/wasm3/issues/160